frontend/components/DownloadButton.vue

```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+
+const { t } = useI18n()
+
+interface Props {
+  value: string
+  cta: string
+  fileName: string
+}
+const props = defineProps<Props>()
+
+const prefix = 'components/DownloadButton'
+const tt = (key: string) => t(`${prefix}.${key}`)
+
+const statePrefix = `${prefix}[${useStateIDGenerator().id()}]`
+const downloaded = useState<boolean>(`${statePrefix}.downloaded`, () => false)
+const message = computed(() => downloaded.value ? tt('Downloaded') : props.cta)
+const icon = computed(() => downloaded.value ? 'pi pi-check' : 'pi pi-download')
+
+const download = async () => {
```

Comment:
nit: doesn't need to be async, there's [a lint for that](https://github.com/typescript-eslint/typescript-eslint/blob/HEAD/packages/eslint-plugin/docs/rules/require-await.md), which we should probably turn on.

Reply:
GC - turned on the lint.